src/object/C4ObjectScript.cpp

```diff
@@ -1703,8 +2119,10 @@ static long FnGetObjectStatus(C4Object *Obj)
 static bool FnAdjustWalkRotation(C4Object *Obj, long iRangeX, long iRangeY, long iSpeed)
 {
 	// must be rotateable and attached to solid ground
-	if (!Obj->Def->Rotateable || ~Obj->Action.t_attach&CNAT_Bottom || Obj->Shape.AttachMat == MNone)
+	if (!Obj->Def->Rotateable || ~(Obj->Action.t_attach & CNAT_Bottom) || Obj->Shape.AttachMat == MNone)
```

Comment:
Those parentheses are incorrect; the negation binds to `Obj->Action.t_attach`, not to the entire masking result.